js/page-specific/swiftype-search-results.js

```diff
@@ -17,52 +17,76 @@ baltimoreCounty.pageSpecific.swiftypeSearchResults = (function swiftypeSearchRes
 		console.error(err);
 	}
 
-	function searchResultRequestSuccessHandler(response) {
-		var info = response.info.page;
-		var hits = response.records.page;
-		var maxPages = 10;
-		var tooManyResults = info.num_pages > maxPages;
-		var lastPage = tooManyResults ? maxPages : info.num_pages;
-		var searchResults = [];
-		var pageLinks = [];
-		var lastPageNumber = info.current_page * info.per_page < info.total_result_count ?
-			info.current_page * info.per_page :
-			info.total_result_count;
-		var firstPageNumber = ((info.current_page - 1) * info.per_page) + 1
-
-		info.base_url = window.location.pathname + '?q=' + info.query + '&page=';
-
-		info.index = {
-			first: firstPageNumber,
-			last: lastPageNumber
-		};
+	function buildSearchResults(hits) {
+		var results = [];
 
 		$.each(hits, function eachHit(index, hit) {
 			var highlight = hit.highlight.body || hit.highlight.sections || hit.highlight.title;
 			var title = hit.title;
 			var url = hit.url;
 
-			searchResults.push({
+			results.push({
 				highlight: highlight,
 				title: title,
 				url: url
 			});
 		});
 
+		return results;
+	}
+
+	function buildPageLinks(lastPage, currentPage) {
+		var pageLinks = [];
+
 		for (var i = 1; i <= lastPage; i += 1) {
 			pageLinks.push({
 				page: i,
-				current: i === info.current_page
+				current: i === currentPage
 			});
 		}
 
+		return pageLinks;
+	}
+
+	function calculateLastResultNumber(info) {
+		return info.current_page * info.per_page < info.total_result_count ?
+			info.current_page * info.per_page :
+			info.total_result_count;
+	}
+
+	function calculateFirstResultNumber(info) {
+		return ((info.current_page - 1) * info.per_page) + 1;
+	}
+
+	function searchResultRequestSuccessHandler(response) {
+		var info = response.info.page;
```

Comment:
Is this a type of object that could be reused. Not sure what it would be, maybe like search settings or search config or something like that?

Reply:
Maybe in the future, but right now it's only used by this module.